en/docs/develop-components/develop-an-external-consumer.md

```diff
@@ -0,0 +1,98 @@
+3. Click the **External Consumer** card. 
+4. In the **Create an External Consumer** pane that opens, enter a display name, a unique name, and a description for the component.
+5. Click **Create**. This creates the component and takes you to its **Overview** page.
+   
+## Step 2: Manage authentication for the external consumer
+
+You can configure the external consumer to work with the Choreo built-in identity provider, Asgardeo, or any external identity provider that supports OIDC/OAuth2.0.
+
+Click the respective tab for details depending on the identity provider you want to configure: 
+
+=== "Choreo built-in identity provider"
+
+     Follow the steps given below to configure the built-in identity provider by generating authentication keys:
+
+    !!! note
+         Choreo built-in identity provider is configured by default. Therefore, this step is optional.
+
+     1. In the **Overview** page of the component, under **Authentication Configurations**, click **Configure** corresponding to the environment for which you want to configure an identity provider.
+     2. In the **Identity Provider** list, select **Choreo Built-In Identity Provider**.
+     3. Click **Generate Keys**. 
+
+        !!! Note
+             If you see **Regenerate Secret** instead of **Generate Keys**, it indicates that OAuth keys are already generated for the component in the selected environment.
+
+=== "Asgardeo"
+
+     **Step 2.1: Create and configure an OIDC/OAuth2.0 application in Asgardeo**
+
+     1. Sign in to [Asgardeo](https://console.asgardeo.io/).
+     2. In the top navigation menu, click the **Organization** list and select your organization.
+     3. In the Asgardeo Console left navigation menu, click **Applications**.
+     4. Click **+ New Application**.
+     5. Click **Standard-Based Application**.
+     6. Specify a name for the application and select **OAuth2.0 OpenID Connect** as the protocol.
+     7. Click **Register**.
+     8. Click the **Protocol** tab and follow these steps:
+
+         1. Select `Code` and `Refresh Token` as the **Allowed grant types**.
+         2. In the **Access Token** section, select `JWT` as the **Token type**.
+         3. Click **Update**. 
+         4. Copy the **Client ID** and **Client Secret** of the application. You will need to use these values in the next step to link the OIDC/OAuth2.0 application to your Choreo component.
+
+     **Step 2.2: Link the OIDC/OAuth2.0 application to the Choreo external consumer component**
+
+     1. In the **Overview** page of the component, under **Authentication Configurations**, click **Configure** corresponding to the environment for which you want to configure an identity provider.
+     2. In the **Identity Provider** list, select **Asgardeo - [your-org-name]**.
+     3. Paste the **Client ID** and **Client Secret** of the OIDC/OAuth2.0 application you created in Asgardeo. 
+     4. Click **Add Keys**.
```

Comment:
Let's remove special casing Asgardeo because, 
1. We are planning to remove the special casing of Asgardeo
2. We can't determine the correct Application template type (Standard-Based Application)

Reply:
Removed content on Asgardeo.